db-engine/src/main/java/com/splicemachine/db/impl/jdbc/EmbedResultSet.java

```diff
@@ -616,6 +619,9 @@ public void close() throws SQLException	{
 
 			// we hang on to theResults and messenger
 			// in case more calls come in on this resultSet
+
+			LanguageConnectionContext lcc = getEmbedConnection().getLanguageConnection();
+			lcc.logEndFetching(getSQLText(), NumberofFetchedRows);
```

Comment:
lcc is accessed here after the call to markClosed(), but getLanguageConnection() expects the connection to be open : 
	public final LanguageConnectionContext getLanguageConnection() {
		if (SanityManager.DEBUG)
			SanityManager.ASSERT(**!isClosed()** || isAborting(), "connection is closed");

... maybe it would be better to move the logEndFetching call into the finally block, above the call to markClosed() at line 612.

Reply:
OK